bokehjs/src/lib/core/util/array.ts

```diff
@@ -150,6 +150,12 @@ export function argmax(array: number[]): number {
   return max_by(range(array.length), (i) => array[i])
 }
 
+export function argsort(array: number[]): number[] {
+  const indices = Array.from(array.keys())
+  indices.sort((a, b) => array[a] - array[b])
```

Comment:
I think this may not be relevant anymore, but this doesn't produce a stable sort with certain underlying sorting algorithm. `sort_by()` considers the case when `array[a] == array[b]` and uses indices instead.